jedi/inference/gradual/typing.py

```diff
@@ -277,6 +314,38 @@ def _get_wrapped_value(self):
             .py__getattribute__('tuple').execute_annotation()
         return tuple_
 
+    def infer_type_vars(self, value_set, is_class_value=False):
+        # Circular
+        from jedi.inference.gradual.annotation import merge_pairwise_generics, merge_type_var_dicts
+        from jedi.inference.gradual.base import GenericClass
+
+        if self._is_homogenous():
```

Comment:
I think this is an illegal access. Feel free to just do a private access. I will then find a way how we get that attribute when I merge the pull request.

Reply:
Not sure I follow what you mean by "illegal access"? Do you mean that `TypingClassValueWithIndex.infer_type_vars` should keep this `'Tuple'` handling directly in that method and we shouldn't have an `infer_type_vars` on `jedi`'s `Tuple` class?

Reply:
Kind of I guess :) I only realize now why this is working for you. I guess this could crash in some cases, so it would probably make sense to move it to `Tuple` or back to `elif annotation_name == 'Tuple'` in `TypingClassValueWithIndex`.

Reply:
I'm missing something -- what's the scenario where this could crash?
Note that this code (i.e: this block of diff we're commenting on) _is_ within the `Tuple` class.

Reply:
LOL. I'm sorry, I really didn't see that. The diff showed me `TypingClassValueWithIndex`. Normally that kind of works, but here it seems to be totally wrong. So my bad.